torch/_subclasses/fake_tensor.py

```diff
@@ -722,6 +839,27 @@ def wrap(e, device=None):
 
             return tree_map(partial(wrap), r)
 
+    def may_turn_const(self, t):
+        return (
+            self.const_tensors and t.numel() <= CONSTANT_NUMEL_LIMIT and not t.is_sparse
+        )
+
+    def invalidate_written_to_constants(self, func, flat_arg_tensors, args, kwargs):
+        any_constant = any(e.constant is not None for e in flat_arg_tensors)
+        if any_constant and get_schema_info(func).is_mutable():
+            schema_info = get_schema_info(func)
+            _, new_kwargs = normalize_function(
+                func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+            )
+            for k, v in new_kwargs.items():
+                k = k if (k != "input" or schema_info.has_argument(k)) else "self"
+                if (
+                    isinstance(v, FakeTensor)
+                    and schema_info.is_mutable(k)
+                    and v.constant is not None
+                ):
+                    self.fake_tensor_converter.invalidate_constant_aliases(v.constant)
```

Comment:
Hmm, I guess technically we still need this logic in case you run this without functionalization?

Reply:
yes, my understanding was functionalization will happen post initial tracing with dynamo